gui/qt/util.py

```diff
@@ -610,61 +610,67 @@ def filter(self, p):
                                 for column in columns]))
 
 
-class ButtonsWidget(QWidget):
-
-    def __init__(self):
-        super(QWidget, self).__init__()
-        self.buttons = []
-
-    def resizeButtons(self):
-        frameWidth = self.style().pixelMetric(QStyle.PM_DefaultFrameWidth)
-        x = self.rect().right() - frameWidth
-        y = self.rect().bottom() - frameWidth
-        for button in self.buttons:
-            sz = button.sizeHint()
-            x -= sz.width()
-            button.move(x, y - sz.height())
-
-    def addButton(self, icon_name, on_click, tooltip):
-        button = QToolButton(self)
-        button.setIcon(QIcon(icon_name))
-        button.setStyleSheet("QToolButton { border: none; hover {border: 1px} pressed {border: 1px} padding: 0px; }")
-        button.setVisible(True)
+class OverlayControlWidget(QWidget):
```

Comment:
Hmm... I wonder if this should just be a pure "mixin" instead.  I'm not exactly clear on how multiple inheritance works in here.  In normal C++ you aren't allowed to multiply inherit from QObjects or QWidgets, are you?

Believe it or now in Python.. a pure mixin class would work here.   Let me know if that's something you'd consider a good idea.

An example of a mixin like this is the `MessageBoxMixin` in this very file...

Reply:
To be clear -- what I'm worried about here is the diamond inheritance pattern which (at least on the C++ side) isn't supported by Qt.

It may work in Python though -- Python <-> C++ qt bindings do a lot of "magic" behind the scenes and perhaps they added support for this using special cases.

Anyway let me know if you would be willing to make this a pure mixin (that is, doesn't inherit from anything but is just designed to be mixed into derived classes to add the extra functionality).

Pure mixins like this don't normally work in C++ easily... but in Python they do due to the very dynamic method lookup employed...

Let me know your thoughts.. :)